File: pages/inthecart.py
from selenium.webdriver.common.by import By
import random

class InTheCart:
    #Getting the delete and checkout buttons
    DELETE_BUTTONS = (By.CLASS_NAME, "cart_quantity_delete")
    CHECKOUT_BUTTON = (By.XPATH, '//div[@class="col-sm-6"]/descendant::a')

    def __init__(self, driver):
        self.driver = driver
    
    def delete_items(self, num_of_items):
        #collecting all the delete buttons
        deletes = self.driver.find_elements(*self.DELETE_BUTTONS)
        #randomly selecting as many delete buttons as specified
        selected_deletes = random.sample(deletes, k=num_of_items)
        #clicking all of the randomly selected delete buttons
        for delete in selected_deletes:
            delete.click()
    
    def proceed_to_checkout(self):
        #clicking the button to proceed to checkout
        checkout_butt = self.driver.find_element(*self.CHECKOUT_BUTTON)
        checkout_butt.click()
        if "https://automationexercise.com/#google_vignette" in self.driver.current_url:
            self.driver.back()
            checkout_butt.click()

File: CandTS/task1/cypress/support/pages/InTheCart.ts
/// <reference types="Cypress" />
class InTheCart {
    DELETE_BUTTONS = ".cart_quantity_delete";
    CHECKOUT_BUTTON = '.btn.btn-default.check_out';
  
    deleteItems(num_of_items: number): void {
        cy.get(this.DELETE_BUTTONS).then(deletes => {
            // randomly select items and delete them
            cy.wrap(deletes).its('length').then(length => {
                const itemsToDelete = Cypress._.take(deletes, Math.min(num_of_items, length));
                cy.wrap(itemsToDelete).each(item => cy.wrap(item).click());
            });
        });
    }

    proceedToCheckout(): void {
        // click the checkout button
        cy.get(this.CHECKOUT_BUTTON).click();
    }
}
export default InTheCart;
